### PR TITLE
Do not use mutable default arguments

### PR DESCRIPTION
**Pull request by MaxNoe**
It is generally a bad idea, especially for the config loading.

For a discussion why, see https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

### CODE REVIEW

File: pyirf/io/io.py [hunk 1]
```diff
@@ -82,9 +84,6 @@ def get_simu_info(filepath, particle_name, config={}):
     cfg['diff_cone'] = simu.max_viewcone_radius
     cfg['gen_gamma'] = -simu.spectral_index
 
-    print(particle_name)
-    print(cfg)
```
[the remark this answers is not in the document]
Comment (MaxNoe) [reply pyirf/io/io.py:86]: Really? Why is that needed?

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: for example to make sure that the file that is being read containes that particle type and that the config is correct

[the remark this answers is not in the document]
Comment (MaxNoe) [reply pyirf/io/io.py:86]: If that is critical, we should verify and raise an error if not. Not rely on some print statements.

[the remark this answers is not in the document]
Comment (MaxNoe) [reply pyirf/io/io.py:86]: It just doesn't make any sense that some function prints one of it's input parameters.

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: sure, at least the particle types I guess...
@vuillaut can maybe add if there is anything else to add, this was added at the very beginning

also I think there is another of those default arguments that should be initialized with None and set to {} inside

[the remark this answers is not in the document]
Comment (MaxNoe) [reply pyirf/io/io.py:86]: I think I checked the whole code base for mutable default arguments

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: what about the config argument in this function? isn't one of those cases?

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: ah sorry

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: I was looking apparently at the old line

[the remark this answers is not in the document]
Comment (MaxNoe) [reply pyirf/io/io.py:86]: Yes and I changed it? Do you not see the change?

[the remark this answers is not in the document]
Comment (HealthyPear) [reply pyirf/io/io.py:86]: sorry I wrote you but the comment wasn't posted right away for some reason

**Review by vuillaut — APPROVED** [review]
Thanks for the tip.

**Pull request merged**